site/app/templates/submission/homework/AutogradingResultsBox.twig

```diff
@@ -2,44 +2,57 @@
-                </p>
+        {% if docker_error %}
+            <p class="red-message">Docker Image not present on machine. Please contact your instructor about this.</p>
+            {% if docker_error_data is not null %}
+                <p>Error Details:</p>
+                <ul>
+                    {% for key, value in docker_error_data %}
+                        <li>{{ key }}: {{ value }}</li>
+                    {% endfor %}
+                </ul>
             {% endif %}
-            <script>
-                checkRefreshPage("{{ check_refresh_submission_url|e('js') }}")
-            </script>
-        {% elseif results is not defined %}
-            <p class="red-message">
-                Something went wrong when grading this submission. Please contact your instructor about this.
-            </p>
-            <script>
-                (() => {
-                    const params = new URLSearchParams(window.location.search);
-                    const loadAttempt = Number(params.get('loadAttempt') ?? '0');
-                    if (loadAttempt < 3) {
-                        params.set('loadAttempt', loadAttempt + 1);
-                        window.location.search = params.toString();
-                    }
-                })();
-            </script>
         {% else %}
-            {# Has results! #}
-            {% if show_incentive_message is defined and show_incentive_message %}
+
+            {% if in_queue or in_progress_grading %}
+                {% if in_progress_grading %}
+                    <p class="red-message">
+                        This submission is currently being graded.
+                    </p>
+                {% else %}
+                    <p class="red-message">
+                        This submission is currently in the queue to be graded. Your submission is number {{ queue_pos }} out of {{ queue_total }}.
+                    </p>
+                {% endif %}
                 <script>
-                    (function() {
-                        $('#incentive_message').show();
+                    checkRefreshPage("{{ check_refresh_submission_url|e('js') }}")
+                </script>
+            {% elseif results is not defined %}
+                <p class="red-message">
+                    Something went wrong when grading this submission. Please contact your instructor about this.
+                </p>
+                <script>
+                    (() => {
+                        const params = new URLSearchParams(window.location.search);
+                        const loadAttempt = Number(params.get('loadAttempt') ?? '0');
+                        if (loadAttempt < 3) {
+                            params.set('loadAttempt', loadAttempt + 1);
+                            window.location.search = params.toString();
+                        }
                     })();
                 </script>
```

Comment:
Is this something we can move to a `.js` file?

Reply:
done